EventListener/SchemaFilterListener.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Bundle\MigrationsBundle\EventListener;
+
+use Doctrine\DBAL\Schema\AbstractAsset;
+use Doctrine\Migrations\Metadata\Storage\TableMetadataStorageConfiguration;
+use Doctrine\Migrations\Tools\Console\Command\DoctrineCommand;
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+
+/**
+ * Acts as a schema filter able that hides the migration metadata table except
+ * when the execution context is that of command inside the migrations
+ * namespace.
+ */
+final class SchemaFilterListener
+{
+    /** @var bool */
+    private $enabled = true;
+
+    /** @param AbstractAsset|string $asset */
+    public function __invoke($asset): bool
+    {
+        if (! $this->enabled) {
+            return true;
+        }
+
+        if ($asset instanceof AbstractAsset) {
+            $asset = $asset->getName();
+        }
+
+        return $asset !== (new TableMetadataStorageConfiguration())->getTableName();
```

Comment:
@christian-kolb You could use a debugger or a simple print statement to see if this gets executed

Reply:
It is executed, the drop statement is still generated with the new version though.

Reply:
Does the comparison on line 33 ever return `true`?

Reply:
@greg0ire With this changes, I get the following output:
![CleanShot 2025-01-11 at 10 15 32@2x](https://github.com/user-attachments/assets/cdb51a8f-7692-43e0-832a-537393c8d30b)

```
php bin/console doc:sch:up --dump-sql
bool(true)
bool(true)
DROP TABLE migration_versions;
```

Reply:
Ah sorry, I didn't spot the `!==`… the real question is: "does it ever return `false`" and it looks like it doesn't. So either `$asset`, which you could `var_dump` is never the name of the migrations table, or there is a subtle difference, like maybe the schema is included in `$asset` and not in `(new TableMetadataStorageConfiguration())->getTableName())`.

Aside, when you post code, can you post text and not an image? It will make it possible for me to use copy/paste.

Reply:
With:
```php
public function __invoke($asset): bool
{
    if (! $this->enabled) {
        return true;
    }

    if ($asset instanceof AbstractAsset) {
        $asset = $asset->getName();
    }

    var_dump($asset);
    var_dump((new TableMetadataStorageConfiguration())->getTableName());
    var_dump($asset !== (new TableMetadataStorageConfiguration())->getTableName());

    return $asset !== (new TableMetadataStorageConfiguration())->getTableName();
}
```

it leads to the following output:

```
php bin/console doc:sch:up --dump-sql
string(25) "messenger_messages_id_seq"
string(27) "doctrine_migration_versions"
bool(true)
string(18) "migration_versions"
string(27) "doctrine_migration_versions"
bool(true)
DROP TABLE migration_versions;
```

Reply:
So the problem is, that it's not getting the table name from the configuration of `doctrine_migrations.yaml`, but only the default one. Which makes sense as it's defined as `new TableMetadataStorageConfiguration()`.

Reply:
Great job, that makes sense. We should find a way to inject the right table name.

Reply:
@christian-kolb I pushed a new version, please try it.